examples/history_vehicles_replacement_for_imitation_learning.py

```diff
@@ -115,7 +119,32 @@ def main(
         assert isinstance(scenario.traffic_history, TrafficHistory)
         logger.debug("working on scenario {}".format(scenario.name))
 
-        veh_missions = scenario.discover_missions_of_traffic_histories()
+        traffic_history: TrafficHistory = scenario.traffic_history
+        assert traffic_history is not None
+        active_vehicles: List[
+            TrafficHistory.TrafficHistoryVehicleTimeSlice
+        ] = traffic_history.vehicle_slice_in_range(
+            exists_at_or_after, ends_before, minimum_history_duration
+        )
+        # e.g. filter: v.end_time - v.start_time > 6 and v.start_time == start_time
+        def custom_filter(vehs: List[TrafficHistory.TrafficHistoryVehicleTimeSlice]):
+            nonlocal exists_at_or_after
+            window = 4
+            return (
+                v
+                for v in vehs
+                if v.average_speed > 3
+                and abs(v.start_time) - exists_at_or_after < window
+            )
+
+        used_vehicles = custom_filter(active_vehicles)
```

Comment:
I note that this filter is useful. I think I could pass it as an argument `history_mission_for_vehicle_slice(..)` to avoid having to go through `traffic_history.vehicle_slice_in_range(..)` first.

Final interface could be shortened to:
```python
veh_missions = {
    str(veh_history_window.vehicle_id): veh_history_window
    for veh_time_slice in scenario.history_missions_for_window(
        exists_at_or_after, ends_before, minimum_window, custom_filter=speed_and_time
    )
}
```

```